addons/mail/static/src/core/common/message.js

```diff
@@ -401,6 +407,23 @@ export class Message extends Component {
     /** @param {HTMLElement} bodyEl */
     prepareMessageBody(bodyEl) {}
 
+    prettifyThreadMentions() {
+        if (this.messageBody.el) {
+            const linkEls = this.messageBody.el.getElementsByTagName("a");
+            for (const linkEl of linkEls) {
+                if (linkEl.classList.contains("o_channel_redirect")) {
+                    const linkText = linkEl.innerHTML.substring(1);
+                    const iconClass = linkEl.classList.contains("o_sub_channel")
+                        ? "fa-comments-o"
+                        : "fa-hashtag";
+                    if (iconClass) {
+                        linkEl.innerHTML = `<i class="fa small-icon ${iconClass}"></i>${linkText}`;
```

Comment:
Would be best to put the expression in XML template and render the temple in JS then assign `innerHTML`.

The gain with this is that this is defined in XML, so this is best when this is also with other templates, and this gives less risk of code injection